src/AutoRest.CSharp/Common/Input/ApiTypes.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.ServiceModel.Rest;
+using System.ServiceModel.Rest.Core;
+using System.ServiceModel.Rest.Internal;
+using System.Threading.Tasks;
+using AutoRest.CSharp.Common.Output.Expressions.KnownValueExpressions.Base;
+using AutoRest.CSharp.Common.Output.Expressions.ValueExpressions;
+using AutoRest.CSharp.Generation.Writers;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Requests;
+using AutoRest.CSharp.Output.Models.Shared;
+
+namespace AutoRest.CSharp.Common.Input
+{
+    internal abstract class ApiTypes
```

Comment:
not sure we shall define this class as internal,  do we allow 3rd party to customize and implement this for their need and plug in that into the codegen in the future?

Reply:
It won't live here long term.  Internal is fine for the medium term until we finish the restructuring.